### PR TITLE
Fixes balloon alerts being offset to the left

### LINKED ISSUE #7823: bubble text doesnt line up
## Reproduction: do anything that has a bubble alert message

image

### PR DESCRIPTION
**Pull request by Penwin0**
## About The Pull Request
Balloon alerts no longer appear far to the left of where they should.
closes #7823 

## Why It's Good For The Game
Bug bad

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Screenshot_1308

</details>

## Changelog
:cl:
fix: Balloon alerts aren't offset to the left anymore
/:cl:

### CODE REVIEW

File: code/datums/chatmessage.dm [hunk 1]
```diff
@@ -492,7 +492,8 @@
 		qdel(src)
 		return
 	//handle color
-	tgt_color = color
+	if(color)
+		tgt_color = color
```
Comment (EvilDragonfiend) [inline code/datums/chatmessage.dm:496]: Should this var have any value if there's no colour?

Comment (Penwin0) [reply code/datums/chatmessage.dm:496]: It'd have the default one for balloon alerts in that case

File: code/datums/chatmessage.dm [hunk 2]
```diff
@@ -459,7 +459,7 @@
 		if(5)
 			return "#[num2hex(c, 2)][num2hex(m, 2)][num2hex(x, 2)]"
 
-/atom/proc/balloon_alert(mob/viewer, text, color = null)
+/atom/proc/balloon_alert(mob/viewer, text, color)
```
Comment (Archanial) [inline code/datums/chatmessage.dm]: Do not remove default null value for this parameter

**Review by Archanial — CHANGES_REQUESTED** [review]

File: code/datums/chatmessage.dm (hunk 2 above)
Comment (BrumalLake) [reply code/datums/chatmessage.dm]: aren't arguments null by default anyways

Comment (Archanial) [reply code/datums/chatmessage.dm]: Explicit is better than implicit

Comment (Penwin0) [inline code/datums/chatmessage.dm]:
```suggestion
/atom/proc/balloon_alert(mob/viewer, text, color = null)
```

Comment (Penwin0) [reply code/datums/chatmessage.dm]: Sure I guess

**Review by PowerfulBacon — APPROVED** [review]

File: code/datums/chatmessage.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/datums/chatmessage.dm]: > aren't arguments null by default anyways

Yes, but if a parameter is optional, it should have a default value specified

**Review by Archanial — APPROVED** [review]

**Pull request merged**